src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,55 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { AuthDto } from './dto/auth.dto';
+import { Users } from './entities/users.entity';
+import * as bcrypt from 'bcryptjs';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectRepository(Users)
+    private readonly AuthRepo: Repository<Users>,
+  ) {}
+
+  async signUp(dto: AuthDto) {
+    const { password, email, role } = dto;
```

Comment:
you will not have role on this stage, please check mocks, dont add something without PM approve, as we split it because of very important reason - we have google auth

Reply:
and better use passport library it is in offisial nest authorisation documentation

Reply:
I removed "role" field in signUp